crates/editor/src/movement.rs

```diff
@@ -49,7 +49,12 @@ pub fn saturating_left(map: &DisplaySnapshot, mut point: DisplayPoint) -> Displa
 /// Returns a column to the right of the current point, wrapping
 /// to the next line if that point is at the end of line.
 pub fn right(map: &DisplaySnapshot, mut point: DisplayPoint) -> DisplayPoint {
-    let max_column = map.line_len(point.row());
+    let line_len = map.line_len(point.row());
+    let max_column = if line_len == 0 {
+        line_len
+    } else {
+        line_len - 1
+    };
```

Comment:
@mrnugget I think this condition is specific to vim's normal mode, as the cursor is rendered as a block over the "next" character (even though in zed it's conceptually just before that character). In both non-vim zed and vim insert mode, you don't want it.

I would probably implement our own version of right in the vim crate, as that keeps the concerns separate, but you could also do this `line_len - 1` check if `self.clip_at_line_ends` is true (which is a global vim uses to signal this to the rest of the editor).

[[Longer term, I'd like to diminish the importance of, or remove clip_at_line_ends; but it is a little hard to understand the conequences of that change]].

Reply:
Done, create a `vim::wrapping_right`